magma/primitives/register.py

```diff
@@ -206,3 +206,9 @@ def __init__(self, T: Kind = None,
         elif (has_reset or has_resetn):
             I = Mux(2, T)()(I, init, reset_select)
         reg.I @= as_bits(I)
+
+
+def register(value, **kwargs):
+    T = type(value)
+    ckt = Register(T, **kwargs)
+    return ckt()(value)
```

Comment:
Let's add a simple test.

Also, we may want to provide the ability to pass through the `name` parameter to the instance instead of the generator (to manually override the instance name)

Reply:
Done. Lmk if the interface is good